File: jungle-core/src/game.rs
use async_trait::async_trait;

use crate::{register_service, service::Service, service_controller::ServiceController};

pub struct Game {}

register_service!(Game);

#[async_trait]
impl Service for Game {
    fn name(&self) -> &'static str {
        "Game"
    }

    async fn run(&mut self, service_ctl: &ServiceController) {
        println!("Hello Jungle!");

        service_ctl.stop();
    }
}

impl Default for Game {
    fn default() -> Self {
        Self {}
    }
}

File: jungle-core/src/game.cpp
// Copyright (C) 2026 pointer-to-bios <pointer-to-bios@outlook.com>
// SPDX-License-Identifier: MIT

#include "jungle/core/game.h"

namespace jungle::core {

Game::Game()
        : Service{type_id::of<Game>()} {}

async::future<> Game::run() { co_return; }

};  // namespace jungle::core
